inference-engine/src/readers/ir_reader/ie_ir_parser.cpp

```diff
@@ -660,48 +660,44 @@ std::shared_ptr<ngraph::Node> V10Parser::XmlDeserializer::createNode(
     }
 
     std::shared_ptr<ngraph::Node> ngraphNode;
+
+    // Find registerd opset
+    auto opsetIt = opsets.find(params.version);
+
     if (isDefaultOpSet(params.version)) {
         // Try to create operation from creators
-        for (const auto& creator : creators) {
-            if (creator->shouldCreate(params.type)) {
-                bool useCreator = false;
-                // Check that opset is registered
-                useCreator |= opsets.find(params.version) == opsets.end();
-                if (!useCreator) {
-                    // Check that creator can create operation with the version from opset
-                    const auto opset = opsets.at(params.version);
-                    // Opset should contains the same version of operation or doesn't contain operation with current type
-                    useCreator |= opset.contains_type(creator->getNodeType()) || !opset.contains_type(params.type);
-                }
-                if (useCreator)
-                    ngraphNode = creator->createLayer(inputs, node, weights, params);
-                break;
-            }
+        auto creatorIt = creators.find(params.type);
+        if (creatorIt != creators.end()) {
+            auto const & creator = creatorIt->second;
+            // Check that opset isn't registered
+            // or opset should contains the same version of operation
+            // or doesn't contain operation with current type
+            if (opsetIt == opsets.end()
+                || opsetIt->second.contains_type(creator->getNodeType())
+                || !opsetIt->second.contains_type(params.type))
+                ngraphNode = creator->createLayer(inputs, node, weights, params);
         }
     }
 
     // Try to create operation from loaded opsets
-    if (!ngraphNode && opsets.count(params.version)) {
-        auto opset = opsets.at(params.version);
-        std::string type = params.type;
-
-        if (type == "Const") {
-            type = "Constant";
-        }
+    if (!ngraphNode && opsetIt != opsets.end()) {
+        auto const & type = params.type == "Const"
+                                ? "Constant"
+                                : params.type;
 
         if (params.version == "opset1") {
             // MVN and ROIPooling were missing in opset1
             if (type == "MVN" || type == "ROIPooling") {
-                opset = opsets.at("opset2");
+                opsetIt = opsets.find("opset2");
```

Comment:
What if after this line `opsetIt == opsets.end()`?

Reply:
I kept the previous logic. I assume the opset2 always is registered.

Reply:
added check with exception generation